lib/screens/stake_screen/stake_screen.dart

```diff
@@ -0,0 +1,340 @@
+                          height: 20,
+                          width: 20,
+                          child: CircularProgressIndicator(
+                            valueColor:
+                                AlwaysStoppedAnimation<Color>(Colors.white),
+                            strokeWidth: 2,
+                          ),
+                        )
+                      : null,
+                  onPressed: (bool) {},
+                  label: 'STAKE',
+                  selected: false,
+                  textStyle: _stakeState == StakesStates.canStake
+                      ? MyStyles.blackMediumTextStyle
+                      : MyStyles.lightWhiteMediumTextStyle,
+                ),
+              ),
+            )
+          ],
+        ),
+        secondChild: SizedBox(
+          width: double.infinity,
+          child: SelectionButton(
+              gradient: gradient,
+              onPressed: (bool) {setState(() {
+                _stakeState = StakesStates.stakePendingApprove;
+                _showToast = true;
+                Future.delayed(Duration(seconds: 3), () {
+                  setState(() {
+                    _stakeState = StakesStates.stakeApproved;
+                  });
+                });
+              });},
+              label: 'STAKE',
+              selected: true,
+              textStyle: MyStyles.blackMediumTextStyle),
+        ),
+        crossFadeState: _stakeState == StakesStates.canStake || _stakeState == StakesStates.stakePendingApprove || _stakeState == StakesStates.stakeApproved
+            ? CrossFadeState.showSecond
+            : CrossFadeState.showFirst,
+        duration: Duration(milliseconds: 150));
+  }
+
+  Container _buildTextField() {
+    return Container(
+      decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(MyStyles.cardRadiusSize),
+          color: Color(MyColors.Button_BG_Black),
+          border: Border.all(color: Colors.white.withOpacity(0.1))),
+      padding: EdgeInsets.symmetric(horizontal: 15),
+      child: TextField(
+        cursorColor: Colors.white,
+        style: MyStyles.lightWhiteMediumTextStyle,
+        decoration: InputDecoration(
+            hintText: '0.00',
+            focusedErrorBorder: InputBorder.none,
+            enabledBorder: InputBorder.none,
+            border: InputBorder.none,
+            disabledBorder: InputBorder.none,
+            focusedBorder: InputBorder.none,
+            suffixIcon: Container(
+              margin: EdgeInsets.all(10),
+              padding: EdgeInsets.symmetric(vertical: 5, horizontal: 8),
+              child: Text(
+                'MAX',
+                style: MyStyles.whiteSmallTextStyle,
+              ),
+              decoration: BoxDecoration(
+                  color: Color(0xFF5BCCBD).withOpacity(0.25),
+                  border: Border.all(color: Colors.white, width: 1.5),
+                  borderRadius: BorderRadius.circular(7)),
+            )),
+      ),
+    );
```

Comment:
Das sieht nach einem Widget aus, das sicher häufiger irgendwo in der App vorkommt oder in Zukunft vorkommen könnte (z.B. beim Lockscreen). 
- [ ] Durchschau mal andere Screens, ob es bereits ähnliches gibt. z.B. den SwapScreen (da hab ich auf jeden Fall auch ein widget mit so nem MAX Button). 
- [ ] Wenn du nichts in die Art findest oder es sich grundlegend unterscheidet, erstell daraus ein neues Widget.